src/agent/TransportService.ts

```diff
@@ -0,0 +1,41 @@
+import { Socket } from 'socket.io';
+
+export class TransportService {
+  private transportTable: TransportTable = {};
+
+  public saveTransport(connectionId: string, transport: Transport) {
+    this.transportTable[connectionId] = transport;
+  }
+
+  public getTransport(connectionId: string) {
+    return this.transportTable[connectionId];
+  }
+}
+
+interface TransportTable {
+  [connectionRecordId: string]: Transport;
+}
+
+type TransportType = 'ws' | 'http';
+
+export interface Transport {
+  type: TransportType;
+}
+
+export class WebSocketTransport implements Transport {
+  public type: TransportType = 'ws';
+  public socket: Socket;
+
+  public constructor(socket: Socket) {
+    this.socket = socket;
+  }
+}
```

Comment:
When the socket (http or ws) will be closed `canReply` can be set to false.

Also when we already replied with a message on the http channel we can also set `canReply` to false